drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTablePresenter.java

```diff
@@ -0,0 +1,1540 @@
+            case EXTENDED_ENTRY:
+                view.newExtendedEntryActionRetractFact();
+                break;
+            case LIMITED_ENTRY:
+                view.newLimitedEntryActionRetractFact();
+                break;
+        }
+    }
+
+    @Override
+    public void newActionWorkItem() {
+        assertNotReadOnly();
+        view.newActionWorkItem();
+    }
+
+    @Override
+    public void newActionWorkItemSetField() {
+        assertNotReadOnly();
+        view.newActionWorkItemSetField();
+    }
+
+    @Override
+    public void newActionWorkItemInsertFact() {
+        assertNotReadOnly();
+        view.newActionWorkItemInsertFact();
+    }
+
+    @Override
+    public void newActionBRLFragment() {
+        assertNotReadOnly();
+        switch ( model.getTableFormat() ) {
+            case EXTENDED_ENTRY:
+                view.newExtendedEntryActionBRLFragment();
+                break;
+            case LIMITED_ENTRY:
+                view.newLimitedEntryActionBRLFragment();
+                break;
+        }
+    }
+
+    @Override
+    public void editCondition( final Pattern52 pattern,
+                               final ConditionCol52 column ) {
+        view.editCondition( pattern,
+                            column );
+    }
+
+    @Override
+    public void editCondition( final BRLConditionColumn column ) {
+        if ( column instanceof LimitedEntryBRLConditionColumn ) {
+            view.editLimitedEntryConditionBRLFragment( (LimitedEntryBRLConditionColumn) column );
+        } else {
+            view.editExtendedEntryConditionBRLFragment( column );
+        }
+    }
+
+    @Override
+    public void editAction( final ActionCol52 column ) {
+        if ( column instanceof ActionWorkItemSetFieldCol52 ) {
+            view.editActionWorkItemSetField( (ActionWorkItemSetFieldCol52) column );
+        } else if ( column instanceof ActionSetFieldCol52 ) {
+            view.editActionSetField( (ActionSetFieldCol52) column );
+        } else if ( column instanceof ActionWorkItemInsertFactCol52 ) {
+            view.editActionWorkItemInsertFact( (ActionWorkItemInsertFactCol52) column );
+        } else if ( column instanceof ActionInsertFactCol52 ) {
+            view.editActionInsertFact( (ActionInsertFactCol52) column );
+        } else if ( column instanceof ActionRetractFactCol52 ) {
+            view.editActionRetractFact( (ActionRetractFactCol52) column );
+        } else if ( column instanceof ActionWorkItemCol52 ) {
+            view.editActionWorkItem( (ActionWorkItemCol52) column );
+        } else if ( column instanceof LimitedEntryBRLActionColumn ) {
+            view.editLimitedEntryActionBRLFragment( (LimitedEntryBRLActionColumn) column );
+        } else if ( column instanceof BRLActionColumn ) {
+            view.editExtendedEntryActionBRLFragment( (BRLActionColumn) column );
+        }
+    }
+
+    @Override
+    public void appendColumn( final AttributeCol52 column ) {
```

Comment:
Some more methods (`appendColumn(..)`) with very similar code that could be reused.